realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -2216,6 +2217,56 @@ private OsResults lazyFindAll() {
         return this;
     }
 
+    /**
+     * Create a text-based predicate using the Realm Query Language. This predicate can be combined
+     * with other raw or type safe predicates.
+     * <p>
+     * Class and property names used in the raw predicate can be either the names defined in the
+     * Realm Model classes or the internal names defined using the {@link io.realm.annotations.RealmClass}
+     * or {@link io.realm.annotations.RealmField} annotations.
+     * </p>
+     * The string-based predicates can contain placeholders that conform to the syntax supported by
+     * {@link java.util.Formatter}.
+     * <p>
+     * See <a href="https://docs.mongodb.com/realm-sdks/js/latest/tutorial-query-language.html">these docs</a>
+     * for a more detailed description of the Realm Query Language.
+     * <p>
+     * <pre>
+     * Examples:
+     * {@code
+     * RealmQuery<Person> query = realm.where(Person.class);
+     *
+     * // Simple query
+     * query.rawPredicate("name = 'Jane'");
+     * query.rawPredicate("name = '%s'", getName());
+     *
+     * // Multiple predicates
+     * query.rawPredicate("name = 'Jane' OR name = 'John'")
+     *
+     * // Collection queries
+     * query.rawPredicate("children.@count > 3")
+     * query.rawPredicate("ALL children.age > 18")
+     *
+     * // Sub queries
+     * query.rawPredicate("SUBQUERY(children, $child, $child.age > 21 AND $child.gender = 'male'").@count > 0');
+     *
+     * // Sort, Distinct, Limit
+     * query.rawPredicate("name = 'Jane' SORT(lastName) DISTINCT(city) LIMIT(5)");
+     * }
+     * </pre>
+     *
+     * @param predicate A Realm Query Language predicate.
+     */
+    public RealmQuery<E> rawPredicate(String predicate, Object... arguments) {
+        realm.checkIfValid();
+        if (Util.isEmptyString(predicate)) {
```

Comment:
You can query on the `"TRUE_PREDICATE"` (AFAIR) if you want all the objects. Don't know if it makes sense to expose that though.

Reply:
Yes, I have been thinking about the same thing. "TRUEPREDICATE" does work as the "select all" option. I don't have strong opinions, but I assumed that if you provided an empty string it was more likely a mistake than not.